### PR TITLE
Move provider examples to doc book

### PR DESCRIPTION
**Pull request by alexander-demicev**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: docs/book/src/03_examples/03_docker.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Cluster API Docker Infrastructure Provider
+
+This page focuses on using the RKE2 provider with the Docker Infrastructure provider.
+
+## Setting up the Management Cluster
+
+Make sure your set up a Management Cluster to use with Cluster API, you can follow instructions from the Cluster API [book](https://cluster-api.sigs.k8s.io/user/quick-start.html).
```
Comment (salasberryfin) [inline docs/book/src/03_examples/03_docker.md:7]:
```suggestion
Make sure you set up a Management Cluster to use with Cluster API, you can follow instructions from the Cluster API [book](https://cluster-api.sigs.k8s.io/user/quick-start.html).
```

File: docs/book/src/03_examples/00.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+## Examples
+
+This section contains examples of how to use the CAPRKE2 with different cloud providers and platforms.
+
+### Setting up the Management Cluster
+
+Make sure your set up a Management Cluster to use with Cluster API, you can follow instructions from the Cluster API [book](https://cluster-api.sigs.k8s.io/user/quick-start.html).
```
Comment (salasberryfin) [inline docs/book/src/03_examples/00.md:7]:
```suggestion
Make sure you set up a Management Cluster to use with Cluster API, you can follow instructions from the Cluster API [book](https://cluster-api.sigs.k8s.io/user/quick-start.html).
```

**Review by salasberryfin — COMMENTED** [review]
Nitpicking a couple of typos but LGTM.

**Comment by furkatgofurov7** [comment]
@alexander-demicev this needs rebase also

**Review by furkatgofurov7 — APPROVED** [review]

**Review by Danil-Grigorev — APPROVED** [review]

**Pull request merged**